Fix for discrete adjoint: axisymmetry + SST turbulence model

lkusch opened the pull request:
## Proposed Changes
The velocity component used in the axisymmetric residual is now considered as an input for preaccumulation in the residual computation. When not considered, the use of axisymmetry and the SST model will result in wrong sensitivities when using the discrete adjoint method.

I also wanted to note that there does not seem to be any treatment of axisymmetric cases for the SA model. Is this correct or does it make sense to include a warning? We could, for example, do that in the context of this pull request. 

## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [x ] I am submitting my contribution to the develop branch.
- [ ] My contribution generates no new compiler warnings (try with the '-Wall -Wextra -Wno-unused-parameter -Wno-empty-body' compiler flags, or simply --warnlevel=2 when using meson).
- [ x] My contribution is commented and consistent with SU2 style.
- [ ] I have added a test case that demonstrates my contribution, if necessary.
- [ ] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp) , if necessary.

pcarruscag commented on SU2_CFD/include/numerics/turbulent/turb_sources.hpp:
```diff
@@ -599,6 +599,7 @@ class CSourcePieceWise_TurbSST final : public CNumerics {
     if (Coord_i[1] < EPS) return;
 
     AD::SetPreaccIn(Coord_i[1]);
+    AD::SetPreaccIn(V_i[2]);
```
Thank you for the fix, this is probably something I broke recently.
Can you add a simple regression test case? It can be a pipe or something very small, just enough to prevent this in the future.

There is an issue opened for the SA problem you mention #1565, we should check for this in CConfig::SetPostprocessing and throw an error (if axisymetry and the turbulence model is not NONE, SST, or SST_SUST).

pcarruscag replied:
And if I could ask one more thing, please replace the hardcoded "2" with
```suggestion
    AD::SetPreaccIn(V_i[idx.Velocity() + 1]);
```
And similarly for uses of `PrimVar_Grad_i` in line 614

pcarruscag commented:
Oh and last thing, add yourself to AUTHORS.md

lkusch commented:
Thanks for the fast reply! I changed the hardcoded 2 (just as a remark: the hardcoded values also appear in the calculations), added myself as an author and inserted some lines for throwing an error if the issue in #1565 occurs. I did not know exactly where to put it best. 

Concerning a regression test: I strongly support the idea of introducing an axisymmetric regression test. However, I was using a testcase from @bigfooted , and I never set up such a test case on my own. There do not seem to be any axisymmetric pipe setups in the Testcases folder so far. @bigfooted , could we maybe use your mesh for the jet flow test case and, if necessary, switch to a standard flow setup?

pcarruscag commented:
Yep that's what I meant, replace it in the calculations as well please, and also in PrimVarGrad, the leading 2 is "velocity + 1" and the leading one is "velocity".
The reason for this replacing is to make the turbulence sources compatible with NEMO.

pcarruscag commented:
We have some small flat plates here and there, I can convert one to axisymmetric.

pcarruscag reviewed (commented):
That's perfect thanks.
I had a look in the testcases and there is an axisym sst regression for the primal solver, in TestCases/axisymmetric_rans/air_nozzle.
It has a restart file already (in the TestCases repo), so it should be easy to add a regression to parallel_regression_AD.py

pcarruscag reviewed (approved):
Everything else LGTM, thanks again.

bigfooted commented:
> Thanks for the fast reply! I changed the hardcoded 2 (just as a remark: the hardcoded values also appear in the calculations), added myself as an author and inserted some lines for throwing an error if the issue in #1565 occurs. I did not know exactly where to put it best.
> 
> Concerning a regression test: I strongly support the idea of introducing an axisymmetric regression test. However, I was using a testcase from @bigfooted , and I never set up such a test case on my own. There do not seem to be any axisymmetric pipe setups in the Testcases folder so far. @bigfooted , could we maybe use your mesh for the jet flow test case and, if necessary, switch to a standard flow setup?

Yes, let's use that mesh. It can be used for pipe flow setup and jets with coflows, so we can reuse it in different testcases as well if needed.

lkusch commented:
I was aware of that test case. But I did not use it for the verification of sensitivities since I could not manage to set up a working derivative computation (different combination of markers/objective functions were leading to a memory error). However, I could use it to set up a regression test for the comparison of residuals.

pcarruscag reviewed (commented):
Where did you get memory errors? SU2_DOT_AD?

TobiKattmann commented on TestCases/axisymmetric_rans/air_nozzle/air_nozzle_adj.cfg:
```diff
@@ -0,0 +1,217 @@
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%                                                                              %
+% SU2 configuration file                                                       %
+% Case description: Axisymmetric supersonic converging-diverging air nozzle    %
+% Author: Florian Dittmann                                                     %
+% Date: 2021.12.02                                                             %
+% File Version 7.3.0 "Blackbird"                                               %
+%                                                                              %
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```
```suggestion
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%                                                                              %
% SU2 configuration file                                                       %
% Case description: Axisymmetric supersonic converging-diverging air nozzle    %
% Author: Lisa Kusch                                                           %
% Date: 2022.23.03                                                             %
% File Version 7.3.0 "Blackbird"                                               %
%                                                                              %
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```

TobiKattmann commented on TestCases/axisymmetric_rans/air_nozzle/air_nozzle_adj.cfg:
```diff
@@ -0,0 +1,217 @@
+%                                                                              %
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+% ------------- DIRECT, ADJOINT, AND LINEARIZED PROBLEM DEFINITION ------------%
+%
+% Physical governing equations (EULER, NAVIER_STOKES,
+%                               FEM_EULER, FEM_NAVIER_STOKES, FEM_RANS, FEM_LES,
+%                               WAVE_EQUATION, HEAT_EQUATION, FEM_ELASTICITY,
+%                               POISSON_EQUATION)
+SOLVER= RANS
+%
+% Specify turbulence model (NONE, SA, SA_NEG, SST, SA_E, SA_COMP, SA_E_COMP)
+KIND_TURB_MODEL= SST
+%
+% Mathematical problem (DIRECT, CONTINUOUS_ADJOINT, DISCRETE_ADJOINT)
+MATH_PROBLEM= DISCRETE_ADJOINT
+%
+% Restart solution (NO, YES)
+RESTART_SOL= NO
+%
+% System of measurements (SI, US)
+% International system of units (SI): ( meters, kilograms, Kelvins,
+%                                       Newtons = kg m/s^2, Pascals = N/m^2,
+%                                       Density = kg/m^3, Speed = m/s,
+%                                       Equiv. Area = m^2 )
+% United States customary units (US): ( inches, slug, Rankines, lbf = slug ft/s^2,
+%                                       psf = lbf/ft^2, Density = slug/ft^3,
+%                                       Speed = ft/s, Equiv. Area = ft^2 )
+SYSTEM_MEASUREMENTS= SI
+%
+AXISYMMETRIC= YES
+%
```
```suggestion
% ------------- DIRECT, ADJOINT, AND LINEARIZED PROBLEM DEFINITION ------------%
%
SOLVER= RANS
KIND_TURB_MODEL= SST
RESTART_SOL= NO
%
AXISYMMETRIC= YES
%
```

TobiKattmann commented on TestCases/axisymmetric_rans/air_nozzle/air_nozzle_adj.cfg:
```diff
@@ -0,0 +1,217 @@
+%
+AXISYMMETRIC= YES
+%
+% -------------------- COMPRESSIBLE FREE-STREAM DEFINITION --------------------%
+%
+% Mach number (non-dimensional, based on the free-stream values)
+MACH_NUMBER= 1E-9
+%
+% Angle of attack (degrees, only for compressible flows)
+AOA= 0.0
+%
+% Side-slip angle (degrees, only for compressible flows)
+SIDESLIP_ANGLE= 0.0
+%
+% Init option to choose between Reynolds (default) or thermodynamics quantities
+% for initializing the solution (REYNOLDS, TD_CONDITIONS)
+INIT_OPTION= TD_CONDITIONS
+%
+% Free-stream option to choose between density and temperature (default) for
+% initializing the solution (TEMPERATURE_FS, DENSITY_FS)
+FREESTREAM_OPTION= TEMPERATURE_FS
+%
+% Free-stream pressure (101325.0 N/m^2, 2116.216 psf by default)
+FREESTREAM_PRESSURE= 1400000
+%
+% Free-stream temperature (288.15 K, 518.67 R by default)
+FREESTREAM_TEMPERATURE= 373.15
+%
+% Compressible flow non-dimensionalization (DIMENSIONAL, FREESTREAM_PRESS_EQ_ONE,
+%                              FREESTREAM_VEL_EQ_MACH, FREESTREAM_VEL_EQ_ONE)
+REF_DIMENSIONALIZATION= DIMENSIONAL
+
```
```suggestion
% -------------------- COMPRESSIBLE FREE-STREAM DEFINITION --------------------%
%
MACH_NUMBER= 1E-9
%
INIT_OPTION= TD_CONDITIONS
%
FREESTREAM_OPTION= TEMPERATURE_FS
FREESTREAM_PRESSURE= 1400000
FREESTREAM_TEMPERATURE= 373.15
%
REF_DIMENSIONALIZATION= DIMENSIONAL
%
```

TobiKattmann commented on TestCases/axisymmetric_rans/air_nozzle/air_nozzle_adj.cfg:
```diff
@@ -0,0 +1,217 @@
+%                              FREESTREAM_VEL_EQ_MACH, FREESTREAM_VEL_EQ_ONE)
+REF_DIMENSIONALIZATION= DIMENSIONAL
+
+% ---- IDEAL GAS, POLYTROPIC, VAN DER WAALS AND PENG ROBINSON CONSTANTS -------%
+%
+% Fluid model (STANDARD_AIR, IDEAL_GAS, VW_GAS, PR_GAS,
+%              CONSTANT_DENSITY, INC_IDEAL_GAS, INC_IDEAL_GAS_POLY)
+FLUID_MODEL= STANDARD_AIR
+
+% --------------------------- VISCOSITY MODEL ---------------------------------%
+%
+% Viscosity model (SUTHERLAND, CONSTANT_VISCOSITY, POLYNOMIAL_VISCOSITY).
+VISCOSITY_MODEL= CONSTANT_VISCOSITY
+%
+% Molecular Viscosity that would be constant (1.716E-5 by default)
+MU_CONSTANT= 1.716E-5 
+
+% --------------------------- THERMAL CONDUCTIVITY MODEL ----------------------%
+%
+% Laminar Conductivity model (CONSTANT_CONDUCTIVITY, CONSTANT_PRANDTL,
+% POLYNOMIAL_CONDUCTIVITY).
+CONDUCTIVITY_MODEL= CONSTANT_PRANDTL
+%
+% Laminar Prandtl number (0.72 (air), only for CONSTANT_PRANDTL)
+PRANDTL_LAM= 0.72
+%
+% Turbulent Prandtl number (0.9 (air) by default)
+PRANDTL_TURB= 0.90
+
```
```suggestion
% ---- IDEAL GAS, POLYTROPIC, VAN DER WAALS AND PENG ROBINSON CONSTANTS -------%
%
FLUID_MODEL= STANDARD_AIR
%
% --------------------------- VISCOSITY MODEL ---------------------------------%
%
VISCOSITY_MODEL= CONSTANT_VISCOSITY
MU_CONSTANT= 1.716E-5 
%
% --------------------------- THERMAL CONDUCTIVITY MODEL ----------------------%
%
CONDUCTIVITY_MODEL= CONSTANT_PRANDTL
PRANDTL_LAM= 0.72
PRANDTL_TURB= 0.90
%
```

TobiKattmann commented on TestCases/axisymmetric_rans/air_nozzle/air_nozzle_adj.cfg:
```diff
@@ -0,0 +1,217 @@
+% Turbulent Prandtl number (0.9 (air) by default)
+PRANDTL_TURB= 0.90
+
+% -------------------- BOUNDARY CONDITION DEFINITION --------------------------%
+%
+% Navier-Stokes (no-slip), constant heat flux wall  marker(s) (NONE = no marker)
+% Format: ( marker name, constant heat flux (J/m^2), ... )
+MARKER_HEATFLUX= ( WALL, 0.0 )
+%
+% Symmetry boundary marker(s) (NONE = no marker)
+MARKER_SYM= ( SYMMETRY )
+%
+% Riemann boundary marker(s) (NONE = no marker)
+% Format: (marker, data kind flag, list of data)
+MARKER_RIEMANN= ( INFLOW, TOTAL_CONDITIONS_PT, 1400000.0, 373.15, 1.0, 0.0, 0.0, OUTFLOW, STATIC_PRESSURE, 100000.0, 0.0, 0.0, 0.0, 0.0 )
+
```
```suggestion
% -------------------- BOUNDARY CONDITION DEFINITION --------------------------%
%
MARKER_HEATFLUX= ( WALL, 0.0 )
MARKER_SYM= ( SYMMETRY )
MARKER_RIEMANN= ( INFLOW, TOTAL_CONDITIONS_PT, 1400000.0, 373.15, 1.0, 0.0, 0.0, \
                  OUTFLOW, STATIC_PRESSURE, 100000.0, 0.0, 0.0, 0.0, 0.0 )
%
```

TobiKattmann commented on TestCases/axisymmetric_rans/air_nozzle/air_nozzle_adj.cfg:
```diff
@@ -0,0 +1,217 @@
+% Format: (marker, data kind flag, list of data)
+MARKER_RIEMANN= ( INFLOW, TOTAL_CONDITIONS_PT, 1400000.0, 373.15, 1.0, 0.0, 0.0, OUTFLOW, STATIC_PRESSURE, 100000.0, 0.0, 0.0, 0.0, 0.0 )
+
+% ------------- COMMON PARAMETERS DEFINING THE NUMERICAL METHOD ---------------%
+%
+% Numerical method for spatial gradients (GREEN_GAUSS, WEIGHTED_LEAST_SQUARES)
+NUM_METHOD_GRAD= GREEN_GAUSS
+%
+% CFL number (initial value for the adaptive CFL number)
+CFL_NUMBER= 1000.0
+%
+% Adaptive CFL number (NO, YES)
+CFL_ADAPT= NO
+%
+% Parameters of the adaptive CFL number (factor down, factor up, CFL min value,
+%                                        CFL max value )
+CFL_ADAPT_PARAM= ( 0.1, 2.0, 10.0, 1000.0 )
+%
+% Maximum Delta Time in local time stepping simulations
+MAX_DELTA_TIME= 1E6
+
```
```suggestion
% ------------- COMMON PARAMETERS DEFINING THE NUMERICAL METHOD ---------------%
%
NUM_METHOD_GRAD= GREEN_GAUSS
%
CFL_NUMBER= 1000.0
CFL_ADAPT= NO
MAX_DELTA_TIME= 1E6
%
```

TobiKattmann commented on TestCases/axisymmetric_rans/air_nozzle/air_nozzle_adj.cfg:
```diff
@@ -0,0 +1,217 @@
+% Maximum Delta Time in local time stepping simulations
+MAX_DELTA_TIME= 1E6
+
+% ----------- SLOPE LIMITER AND DISSIPATION SENSOR DEFINITION -----------------%
+%
+% Monotonic Upwind Scheme for Conservation Laws (TVD) in the flow equations.
+%           Required for 2nd order upwind schemes (NO, YES)
+MUSCL_FLOW= YES
+%
+% Slope limiter (NONE, VENKATAKRISHNAN, VENKATAKRISHNAN_WANG,
+%                BARTH_JESPERSEN, VAN_ALBADA_EDGE)
+SLOPE_LIMITER_FLOW= NONE
+
```
```suggestion
% ----------- SLOPE LIMITER AND DISSIPATION SENSOR DEFINITION -----------------%
%
MUSCL_FLOW= YES
SLOPE_LIMITER_FLOW= NONE
%
```

TobiKattmann commented on TestCases/axisymmetric_rans/air_nozzle/air_nozzle_adj.cfg:
```diff
@@ -0,0 +1,217 @@
+
+% ------------------------ LINEAR SOLVER DEFINITION ---------------------------%
+%
+% Linear solver or smoother for implicit formulations (BCGSTAB, FGMRES, SMOOTHER_JACOBI,
+%                                                      SMOOTHER_ILU, SMOOTHER_LUSGS,
+%                                                      SMOOTHER_LINELET)
+LINEAR_SOLVER= FGMRES
+%
+% Preconditioner of the Krylov linear solver (ILU, LU_SGS, LINELET, JACOBI)
+LINEAR_SOLVER_PREC= ILU
+%
+% Linear solver ILU preconditioner fill-in level (0 by default)
+LINEAR_SOLVER_ILU_FILL_IN= 0
+%
+% Minimum error of the linear solver for implicit formulations
+LINEAR_SOLVER_ERROR= 0.01
+%
+% Max number of iterations of the linear solver for the implicit formulation
+LINEAR_SOLVER_ITER= 10
```
```suggestion
LINEAR_SOLVER= FGMRES
LINEAR_SOLVER_PREC= ILU
LINEAR_SOLVER_ILU_FILL_IN= 0
LINEAR_SOLVER_ERROR= 0.01
LINEAR_SOLVER_ITER= 10
```

TobiKattmann commented on TestCases/axisymmetric_rans/air_nozzle/air_nozzle_adj.cfg:
```diff
@@ -0,0 +1,217 @@
+
+% -------------------------- MULTIGRID PARAMETERS -----------------------------%
+%
+% Multi-grid levels (0 = no multi-grid)
+MGLEVEL= 0
+
```
```suggestion
MGLEVEL= 0
%
```

TobiKattmann commented on TestCases/axisymmetric_rans/air_nozzle/air_nozzle_adj.cfg:
```diff
@@ -0,0 +1,217 @@
+% Multi-grid levels (0 = no multi-grid)
+MGLEVEL= 0
+
+% -------------------- FLOW NUMERICAL METHOD DEFINITION -----------------------%
+%
+% Convective numerical method (JST, LAX-FRIEDRICH, CUSP, ROE, AUSM, AUSMPLUSUP, AUSMPLUSUP2, HLLC,
+%                              TURKEL_PREC, MSW, FDS)
+CONV_NUM_METHOD_FLOW= ROE
+%
+% Entropy fix coefficient (0.0 implies no entropy fixing, 1.0 implies scalar
+%                          artificial dissipation)
+ENTROPY_FIX_COEFF= 0.1
+%
+% Time discretization (RUNGE-KUTTA_EXPLICIT, EULER_IMPLICIT, EULER_EXPLICIT)
+TIME_DISCRE_FLOW= EULER_IMPLICIT
+%
+
```
```suggestion
% -------------------- FLOW NUMERICAL METHOD DEFINITION -----------------------%
%
CONV_NUM_METHOD_FLOW= ROE
ENTROPY_FIX_COEFF= 0.1
TIME_DISCRE_FLOW= EULER_IMPLICIT
%
```

TobiKattmann commented on TestCases/axisymmetric_rans/air_nozzle/air_nozzle_adj.cfg:
```diff
@@ -0,0 +1,217 @@
+TIME_DISCRE_FLOW= EULER_IMPLICIT
+%
+
+% -------------------- TURBULENT NUMERICAL METHOD DEFINITION ------------------%
+%
+% Convective numerical method (SCALAR_UPWIND)
+CONV_NUM_METHOD_TURB= SCALAR_UPWIND
+%
+% Time discretization (EULER_IMPLICIT)
+TIME_DISCRE_TURB= EULER_IMPLICIT
+%
+% Reduction factor of the CFL coefficient in the turbulence problem
+CFL_REDUCTION_TURB= 1.0
+
```
```suggestion
% -------------------- TURBULENT NUMERICAL METHOD DEFINITION ------------------%
%
CONV_NUM_METHOD_TURB= SCALAR_UPWIND
TIME_DISCRE_TURB= EULER_IMPLICIT
CFL_REDUCTION_TURB= 1.0
%
```

TobiKattmann commented on TestCases/axisymmetric_rans/air_nozzle/air_nozzle_adj.cfg:
```diff
@@ -0,0 +1,217 @@
+% Reduction factor of the CFL coefficient in the turbulence problem
+CFL_REDUCTION_TURB= 1.0
+
+% --------------------------- CONVERGENCE PARAMETERS --------------------------%
+%
+% Number of total iterations
+ITER= 1000
+%
+% Min value of the residual (log10 of the residual)
+CONV_RESIDUAL_MINVAL= -12
+%
+% Start convergence criteria at iteration number
+CONV_STARTITER= 10
+
```
```suggestion
% --------------------------- CONVERGENCE PARAMETERS --------------------------%
%
ITER= 1000
CONV_RESIDUAL_MINVAL= -12
CONV_STARTITER= 10
%
```

TobiKattmann commented on TestCases/axisymmetric_rans/air_nozzle/air_nozzle_adj.cfg:
```diff
@@ -0,0 +1,217 @@
+% Start convergence criteria at iteration number
+CONV_STARTITER= 10
+
+% ------------------------- INPUT/OUTPUT INFORMATION --------------------------%
+%
+% Mesh input file
+MESH_FILENAME= nozzle.su2
+%
+% Mesh input file format (SU2, CGNS)
+MESH_FORMAT= SU2
+%
+% Restart flow input file
+SOLUTION_FILENAME= solution_flow.dat
+%
+% Restart adjoint input file
+SOLUTION_ADJ_FILENAME= solution_adj.dat
+%
+% Output file restart adjoint
+RESTART_ADJ_FILENAME= restart_adj.dat
+%
+% Writing solution file frequency
+OUTPUT_WRT_FREQ= 1000
+%
+% Screen output
+SCREEN_OUTPUT= (INNER_ITER, RMS_ADJ_DENSITY, RMS_ADJ_ENERGY, RMS_ADJ_TKE, RMS_ADJ_DISSIPATION)
```
```suggestion
% ------------------------- INPUT/OUTPUT INFORMATION --------------------------%
%
MESH_FILENAME= nozzle.su2
%
SOLUTION_FILENAME= solution_flow.dat
SOLUTION_ADJ_FILENAME= solution_adj.dat
RESTART_ADJ_FILENAME= restart_adj.dat
%
OUTPUT_WRT_FREQ= 1000
%
SCREEN_OUTPUT= (INNER_ITER, RMS_ADJ_DENSITY, RMS_ADJ_ENERGY, RMS_ADJ_TKE, RMS_ADJ_DISSIPATION)
```

TobiKattmann commented on TestCases/parallel_regression_AD.py:
```diff
@@ -168,6 +168,21 @@ def main():
+    discadj_axisymmetric_rans_nozzle.cfg_file  = "air_nozzle_adj.cfg"
+    discadj_axisymmetric_rans_nozzle.test_iter = 10
+    discadj_axisymmetric_rans_nozzle.test_vals = [10.000000, -10.391857, -15.524696, -7.715907, -17.350541]        
+    discadj_axisymmetric_rans_nozzle.su2_exec  = "parallel_computation.py -f"
```
```suggestion
    discadj_axisymmetric_rans_nozzle.su2_exec  = "mpirun -n 2 SU2_CFD_AD"
    discadj_axisymmetric_rans_nozzle.no_restart = True
```

TobiKattmann replied:
I removed `MATH_PROBLEM= DISCRETE_ADJOINT` in my suggestion above, using SU2_CFD_AD directly ensures that a DA run is done. parallel_computation (to my knowledge) also executes SU2_SOL after SU2_CFD which is the major difference to this

TobiKattmann commented on TestCases/parallel_regression_AD.py:
```diff
@@ -168,6 +168,21 @@ def main():
+    discadj_axisymmetric_rans_nozzle.cfg_dir   = "axisymmetric_rans/air_nozzle"
+    discadj_axisymmetric_rans_nozzle.cfg_file  = "air_nozzle_adj.cfg"
+    discadj_axisymmetric_rans_nozzle.test_iter = 10
+    discadj_axisymmetric_rans_nozzle.test_vals = [10.000000, -10.391857, -15.524696, -7.715907, -17.350541]        
```
```suggestion
    discadj_axisymmetric_rans_nozzle.test_vals = [-10.391857, -15.524696, -7.715907, -17.350541]        
```
Not necessary to test the  iteration number (only the last entries of the screen output are compared)

lkusch commented:
Sorry for the confusion. I must have mixed something up yesterday - the test case was crashing directly for SU2_CFD_AD when I set MARKER_MONITORING. But now it is working. So I think we can stick to the regression case and the way I set it up. Thanks again for the reviews!

lkusch commented:
This is confusing me a bit. Before I set it up: Do you want me to use the original config (without the _adj), remove the comments and the model and then run SU2_CFD_AD in the regression?

TobiKattmann commented:
[27d35a2](https://github.com/su2code/SU2/pull/1571/commits/27d35a2dea4948c00cbb22e1fd88c0feea24f9ab) -> awesome.

Concerning the option(-comment) stripping; I am still in favor, although @FlorianDm  is the original author. Especially I would strip the `AOA, SIDESLIP_ANGLE, CFD_ADAPT_PARAM, SYSTEM_MEASUREMENTS, MESH_FORMAT` as they seem superfluous to me.

But for the sake of keeping this PR concise the current state is fine as well :+1:
